agent/discovery/sapsystem.go

```diff
@@ -39,27 +39,29 @@ func (d SAPSystemsDiscovery) Discover() error {
 		if err != nil {
 			return err
 		}
+	}
 
-		// Store SAP System, Landscape and Environment names on hosts metadata
-		err = storeSAPSystemTags(d.discovery.client, s)
-		if err != nil {
-			return err
-		}
+	// Store SAP System, Landscape and Environment names on hosts metadata
+	err = storeSAPSystemTags(d.discovery.client, d.SAPSystems)
+	if err != nil {
+		return err
 	}
 
 	return nil
 }
 
-func storeSAPSystemTags(client consul.Client, system *sapsystem.SAPSystem) error {
-	envName, landName, sysName, err := loadSAPSystemTags(client, system.SID)
+func storeSAPSystemTags(client consul.Client, systems sapsystem.SAPSystemsList) error {
+	envName, landName, _, err := loadSAPSystemTags(client, systems[0].SID)
```

Comment:
The `loadSAPSystemTags` is only used here, so we could even remove this 3rd return value (which is not used)